test/Sentry.Tests/SentryFailedRequestHandlerTests.cs

```diff
@@ -0,0 +1,294 @@
+
+        // Act
+        sut.CaptureEvent(request, response);
+
+        // Assert
+        _hub.DidNotReceive().CaptureEvent(Arg.Any<SentryEvent>());
+    }
+
+    [Fact]
+    public void CaptureEvent_NoMatchingTarget_DontCapture()
+    {
+        // Arrange
+        var options = new SentryOptions
+        {
+            CaptureFailedRequests = true,
+            FailedRequestTargets = new List<SubstringOrRegexPattern> { "http://foo/" }
+        };
+        var sut = GetSut(options);
+
+        var request = new HttpRequestMessage(HttpMethod.Get, "http://bar/"); 
+        var response = InternalServerErrorResponse();
+
+        // Act
+        sut.CaptureEvent(request, response);
+
+        // Assert
+        _hub.DidNotReceive().CaptureEvent(Arg.Any<SentryEvent>());
+    }
+
+    [Fact]
+    public void CaptureEvent_Capture_FailedRequest()
+    {
+        // Arrange
+        var options = new SentryOptions
+        {
+            CaptureFailedRequests = true
+        };
+        var sut = GetSut(options);
+
+        var request = new HttpRequestMessage();
+        var response = InternalServerErrorResponse(); 
+
+        // Act
+        sut.CaptureEvent(request, response);
+
+        // Assert
+        _hub.Received(1).CaptureEvent(Arg.Any<SentryEvent>(), Arg.Any<Scope>());
+    }
+
+    [Fact]
+    public void CaptureEvent_Capture_Mechanism()
+    {
+        throw new NotImplementedException();
+    }
+
+    [Fact]
+    public void CaptureEvent_Capture_Hints()
+    {
+        throw new NotImplementedException();
+
+        //verify(fixture.hub).captureEvent(
+        //    any(),
+        //    check<Hint> {
+        //    assertNotNull(it.get(TypeCheckHint.OKHTTP_REQUEST))
+        //        assertNotNull(it.get(TypeCheckHint.OKHTTP_RESPONSE))
+        //    }
+        //)
+    }
+
+    [Fact]
+    public void CaptureEvent_Capture_RequestAndResponse()
+    {
+        // Arrange
+        var options = new SentryOptions
+        {
+            CaptureFailedRequests = true
+        };
+        var sut = GetSut(options);
+
+        var url = "http://foo/bar/hello";
+        var queryString = "myQuery=myValue";
+        var fragment = "myFragment";
+        var request = new HttpRequestMessage(HttpMethod.Get, $"{url}?{queryString}#{fragment}");
+        var response = InternalServerErrorResponse(); // This is in the range
+
+        // Act
+        SentryEvent @event = null;
+        _hub.CaptureEvent(Arg.Do<SentryEvent>(e => @event = e));
+        sut.CaptureEvent(request, response);
+
+        // Assert
+        using (new AssertionScope())
+        {            
+            @event.Should().NotBeNull();
+
+            @event.Request.Method.Should().Be(HttpMethod.Post.ToString());
+            @event.Request.Url.Should().Be(url);
+            @event.Request.QueryString.Should().Be(queryString);
+            // Not sure how we want to handle fragments in .NET... potentially we add a Fragment
```

Comment:
The comment says it all here:
Not sure how we want to handle fragments in .NET... potentially we add a Fragment property to the Request class, but that might be a breaking change.

Other options:

We don't capture the fragment specifically (just leave it part of the queryString)... but that would be inconsistent with the Java API
We could put the fragment in the Request.Other collection (as shown in the code that's commented out below).

I think the cleanest and most consistent option would be to add a Fragment property to the Request class but that's a losely held opinion.

Reply:
Fragments are constrained to the browser by design of URLs. They are not sent over HTTP at all.

Reply:
I see it in the Java code.  TBH, I have no clue why they did that there.  Maybe something specific to Android or Java?  AFAIK, fragments don't go over HTTP - period.  They're for the browser only.

Reply:
I'll check with the Java team, but I think their scrubbing of fragments was unnecessary.  They should already be absent.  Per the HTTP spec, HTTP works with URLs.  Fragments are part of URIs, but not URLs, and they're disallowed over HTTP.

Reply:
Confirmed. We don't need to worry about fragments.  They will never be present.  This was a misunderstanding, and shouldn't be in the Java SDK either, nor in the Sentry protocol (`event.request.fragment`). I'll look to have it deprecated.